data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/SqsWorker.java

```diff
@@ -172,9 +172,13 @@ private List<DeleteMessageBatchRequestEntry> processSqsEvents(final List<Message
         }
 
         for (Message message : messages) {
-            Duration duration = Duration.between(Instant.ofEpochMilli(Long.parseLong(message.attributes().get(MessageSystemAttributeName.SENT_TIMESTAMP))), Instant.now());
-            sqsMessageDelayTimer.record(duration.isNegative() ? Duration.ZERO : duration); // Negative durations can occur if messages are processed immediately
-
+            final int receiveCount = Integer.parseInt(message.attributes().get(MessageSystemAttributeName.APPROXIMATE_RECEIVE_COUNT));
+            if (receiveCount < 1) {
```

Comment:
This should be `receiveCount <= 1`, since the receive count will always be 1 when it's grabbed for the first time.

Reply:
good catch, changing this right away